server/route.go

```diff
@@ -1451,7 +1449,7 @@ func (c *client) processRemoteSub(argo []byte, hasOrigin bool) (err error) {
 		sub.sid = append(sub.sid, ' ')
 		sub.sid = append(sub.sid, sub.subject...)
 	}
-	key := string(sub.sid)
+	key := bytesToString(sub.sid)
```

Comment:
Going back to client.go too, we trade here doing a copy of the `sub.sid` but then we reference the RSUB protocol. I guess we do it for the sub.subject/queue, etc.., so maybe it's ok.

Reply:
I think this one ok.